zebra-chain/src/commitments/sapling/arbitrary.rs

```diff
@@ -0,0 +1,33 @@
+use proptest::{arbitrary::any, array, prelude::*};
+
+use crate::commitments::sapling;
+
+impl Arbitrary for sapling::NoteCommitment {
+    type Parameters = ();
+
+    fn arbitrary_with(_args: Self::Parameters) -> Self::Strategy {
+        array::uniform32(any::<u8>())
+            .prop_filter("Valid jubjub::AffinePoint", |b| {
+                jubjub::AffinePoint::from_bytes(*b).is_some().unwrap_u8() == 1
+            })
+            .prop_map(Self::from)
```

Comment:
Rather than doing trial decompression, checking success, and then redoing the decompression, this could save computation by only decompressing once.

But this is downstream of the representation/conversion discussions above.